demos/projects/ESPRESSIF/esp32/components/sample-azure-iot/transport_tls_esp32.c

```diff
@@ -0,0 +1,188 @@
+ * @brief TLS transport interface implementations. This implementation uses
+ * mbedTLS.
+ */
+
+/* Standard includes. */
+#include <string.h>
+
+/* FreeRTOS includes. */
+#include "freertos/FreeRTOS.h"
+
+/* TLS transport header. */
+#include "transport_tls_socket.h"
+
+#include "esp_log.h"
+
+/* TLS includes. */
+#include "esp_transport_ssl.h"
+
+/**
+ * @brief Definition of the network context for the transport interface
+ * implementation that uses mbedTLS and FreeRTOS+TLS sockets.
+ */
+struct NetworkContext
+{
+    esp_transport_handle_t transport;
+    uint32_t receiveTimeoutMs;
+    uint32_t sendTimeoutMs;
+};
+
+static const char *TAG = "tls_freertos";
+/*-----------------------------------------------------------*/
+
+TlsTransportStatus_t TLS_Socket_Connect( NetworkContext_t * pNetworkContext,
+                                           const char * pHostName,
+                                           uint16_t port,
+                                           const NetworkCredentials_t * pNetworkCredentials,
+                                           uint32_t receiveTimeoutMs,
+                                           uint32_t sendTimeoutMs )
+{
+    TlsTransportStatus_t returnStatus = eTLSTransportSuccess;
+
+    if( ( pNetworkContext == NULL ) ||
+        ( pHostName == NULL ) ||
+        ( pNetworkCredentials == NULL ) )
+    {
+        ESP_LOGE(TAG, "Invalid input parameter(s): Arguments cannot be NULL. pNetworkContext=%p, "
+                    "pHostName=%p, pNetworkCredentials=%p.",
+                    pNetworkContext,
+                    pHostName,
+                    pNetworkCredentials);
+        return eTLSTransportInvalidParameter;
+    }
+    
+    pNetworkContext->transport = esp_transport_ssl_init();
+    pNetworkContext->receiveTimeoutMs = receiveTimeoutMs;
+    pNetworkContext->sendTimeoutMs = sendTimeoutMs;
+    if (pNetworkCredentials->ppcAlpnProtos) {
+        esp_transport_ssl_set_alpn_protocol(pNetworkContext->transport, pNetworkCredentials->ppcAlpnProtos);
+    }
+
+    if (pNetworkCredentials->xDisableSni) {
+        esp_transport_ssl_skip_common_name_check(pNetworkContext->transport);
+    }
+
+    if (pNetworkCredentials->pucRootCa) {
+        esp_transport_ssl_set_cert_data_der(pNetworkContext->transport, (const char *)pNetworkCredentials->pucRootCa, pNetworkCredentials->xRootCaSize);
+    }
+
+    if (pNetworkCredentials->pucClientCert) {
+        esp_transport_ssl_set_client_cert_data_der(pNetworkContext->transport, (const char *)pNetworkCredentials->pucClientCert, pNetworkCredentials->xClientCertSize);
+    }
+
+    if (pNetworkCredentials->pucPrivateKey) {
+        esp_transport_ssl_set_client_key_data_der(pNetworkContext->transport, (const char *)pNetworkCredentials->pucPrivateKey, pNetworkCredentials->xPrivateKeySize);
+    }
+
+    if (esp_transport_connect(pNetworkContext->transport, pHostName, port, receiveTimeoutMs) < 0) {
+        returnStatus = eTLSTransportConnectFailure;
+    } else {
```

Comment:
can we please use next line for braces, this file right now is mix with placing braces.